vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListDataProviderRefreshIT.java

```diff
@@ -0,0 +1,60 @@
+package com.vaadin.flow.component.virtuallist.tests;
+
+import com.vaadin.flow.component.virtuallist.testbench.VirtualListElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+import com.vaadin.tests.AbstractComponentIT;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+@TestPath("vaadin-virtual-list/data-provider-refresh")
+public class VirtualListDataProviderRefreshIT extends AbstractComponentIT {
+    private VirtualListElement virtualList;
+    private TestBenchElement updateItemFive;
+    private TestBenchElement updateAllItems;
+
+    @Before
+    public void init() {
+        open();
+        virtualList = $(VirtualListElement.class).waitForFirst();
+        updateItemFive = $("button").id("update-item-five");
+        updateAllItems = $("button").id("update-all-items");
+    }
+
+    @Test
+    public void refreshItem_itemUpdated() {
+        // Wait for initial renderer
+        virtualList.$("flow-component-renderer").waitForFirst();
+        // Check initial render
+        assertItemText(4, "Item5");
+
+        updateItemFive.click();
+
+        // Check updated render
+        assertItemText(4, "Item5 updated");
+    }
+
+    @Test
+    public void refreshAll_allItemsUpdated() {
```

Comment:
This was working before as it uses different methods in the connector, but couldn't hurt to have a test for this as well.